crates/oxc_linter/src/rules/eslint/func_names.rs

```diff
@@ -0,0 +1,507 @@
+use oxc_ast::ast::{
+    AssignmentTarget, AssignmentTargetProperty, BindingPatternKind, Expression, Function,
+    FunctionType, PropertyKind,
+};
+use oxc_ast::AstKind;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{Atom, Span};
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Default, Clone)]
+pub struct FuncNames {
+    default_config: FuncNamesConfig,
+    generators_config: FuncNamesConfig,
+}
+
+#[derive(Debug, Default, Clone, PartialEq)]
+enum FuncNamesConfig {
+    #[default]
+    Always,
+    AsNeeded,
+    Never,
+}
+
+impl FuncNamesConfig {
+    pub fn from(raw: &str) -> Self {
+        match raw {
+            "always" => FuncNamesConfig::Always,
+            "as-needed" => FuncNamesConfig::AsNeeded,
+            "never" => FuncNamesConfig::Never,
+            _ => FuncNamesConfig::default(),
+        }
+    }
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Require or disallow named function expressions
+    ///
+    /// ### Why is this bad?
+    ///
+    /// If you leave off the function name then when the function throws an exception you are likely
```

Comment:
Please improve this documentation. Consider copying it from the ESLint website.

Reply:
By replacing the first line with, for example, "Leaving the name off a function will cause `<anonymous>` to appear in stack traces of errors thrown in it or any function called within it. This makes it more difficult to find where an error is thrown."